[CI] EsqlActionTaskIT testCancelRequestWhenFailingFetchingPages failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #5010 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/rrchati7etfvi)
- [elasticsearch-periodic-platform-support #5010 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/p3gc2y4rbt4a2)
- [elasticsearch-periodic #5032 / graalvm-ce17_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/7ae6caz6kqwdu)
- [elasticsearch-periodic-platform-support #5004 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/7apd62hzil3iw)
- [elasticsearch-periodic #5026 / openjdk23_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/g3xyt5eyxslig)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.EsqlActionTaskIT.testCancelRequestWhenFailingFetchingPages" -Dtests.seed=624E7DA82F780A10 -Dtests.locale=en-CA -Dtests.timezone=Africa/Abidjan -Druntime.java=23
```

**Applicable branches:**
8.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.EsqlActionTaskIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testCancelRequestWhenFailingFetchingPages),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a value greater than <0>
     but: <0> was equal to <0>
```

**Issue Reasons:**
- [8.17] 5 failures in test testCancelRequestWhenFailingFetchingPages (5.2% fail rate in 97 executions)
- [8.17] 2 failures in pipeline elasticsearch-periodic-platform-support (50.0% fail rate in 4 executions)
- [8.17] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -232,9 +232,6 @@ tests:
 - class: org.elasticsearch.xpack.inference.DefaultEndPointsIT
   method: testInferDeploysDefaultElser
   issue: https://github.com/elastic/elasticsearch/issues/114913
-- class: org.elasticsearch.xpack.esql.action.EsqlActionTaskIT
-  method: testCancelRequestWhenFailingFetchingPages
-  issue: https://github.com/elastic/elasticsearch/issues/117397
 - class: org.elasticsearch.xpack.security.operator.OperatorPrivilegesIT
   method: testEveryActionIsEitherOperatorOnlyOrNonOperator
   issue: https://github.com/elastic/elasticsearch/issues/102992

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EsqlActionTaskIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EsqlActionTaskIT.java
@@ -392,12 +392,13 @@ public class EsqlActionTaskIT extends AbstractPausableIntegTestCase {
                 .get();
             ensureYellowAndNoInitializingShards("test");
             request.query("FROM test | LIMIT 10");
-            request.pragmas(randomPragmas());
+            QueryPragmas pragmas = randomPragmas();
+            request.pragmas(pragmas);
             PlainActionFuture<EsqlQueryResponse> future = new PlainActionFuture<>();
             client.execute(EsqlQueryAction.INSTANCE, request, future);
             ExchangeService exchangeService = internalCluster().getInstance(ExchangeService.class, dataNode);
-            boolean waitedForPages;
-            final String sessionId;
+            final boolean waitedForPages;
+            final String exchangeId;
             try {
                 List<TaskInfo> foundTasks = new ArrayList<>();
                 assertBusy(() -> {
@@ -411,13 +412,22 @@ public class EsqlActionTaskIT extends AbstractPausableIntegTestCase {
                     assertThat(tasks, hasSize(1));
                     foundTasks.addAll(tasks);
                 });
-                sessionId = foundTasks.get(0).taskId().toString();
+                final String sessionId = foundTasks.get(0).taskId().toString();
                 assertTrue(fetchingStarted.await(1, TimeUnit.MINUTES));
-                String exchangeId = exchangeService.sinkKeys().stream().filter(s -> s.startsWith(sessionId)).findFirst().get();
+                List<String> sinkKeys = exchangeService.sinkKeys()
+                    .stream()
+                    .filter(
+                        s -> s.startsWith(sessionId)
+                            // exclude the node-level reduction sink
+                            && s.endsWith("[n]") == false
+                    )
+                    .toList();
+                assertThat(sinkKeys.toString(), sinkKeys.size(), equalTo(1));
+                exchangeId = sinkKeys.get(0);
                 ExchangeSinkHandler exchangeSink = exchangeService.getSinkHandler(exchangeId);
                 waitedForPages = randomBoolean();
                 if (waitedForPages) {
-                    // do not fail exchange requests until we have some pages
+                    // do not fail exchange requests until we have some pages.
                     assertBusy(() -> assertThat(exchangeSink.bufferSize(), greaterThan(0)));
                 }
             } finally {
@@ -429,7 +439,7 @@ public class EsqlActionTaskIT extends AbstractPausableIntegTestCase {
             // As a result, the exchange sinks on data-nodes won't be removed until the inactive_timeout elapses, which is
             // longer than the assertBusy timeout.
             if (waitedForPages == false) {
-                exchangeService.finishSinkHandler(sessionId, failure);
+                exchangeService.finishSinkHandler(exchangeId, failure);
             }
         } finally {
             transportService.clearAllRules();

